lib/Excel/Writer/XLSX/Worksheet.pm

```diff
@@ -3333,15 +3333,20 @@ sub data_validation {
         $param->{validate} = $valid_type{ lc( $param->{validate} ) };
     }
 
-
-    # No action is required for validation type 'any'.
-    # TODO: we should perhaps store 'any' for message only validations.
-    return 0 if $param->{validate} eq 'none';
-
-
-    # The list and custom validations don't have a criteria so we use a default
-    # of 'between'.
-    if ( $param->{validate} eq 'list' || $param->{validate} eq 'custom' ) {
+    # No action is required for validation type 'any'
+    # unless there are input messages.
+    return 0
```

Comment:
Convert this from a postfix if() to a standard if(). The code base only uses postfix if() for single items.